lib/shell/shellscript.js

```diff
@@ -59,12 +60,18 @@ class ShellScriptWorker {
     }
 }
 
-function prepareErrorFile(directory) {
+async function prepareErrorFile(directory) {
     const errDir = path.resolve(directory, "errors");
-    fs.mkdirSync(errDir, {recursive: true});
+    await fs.mkdirs(errDir, {recursive: true});
     return path.resolve(errDir, "error.json");
 }
 
+async function prepareMimeInfoFile(directory, renditionId) {
+    const mimeDir = path.resolve(directory, "mimeinfo");
```

Comment:
Maybe instead of extra directories have one common directory for these special files: `metadata`

And put the error and type file in there.

Since clients are supposed to use the path from the env var we should be able to make this change for the error file.

Reply:
Can we do this in a second step? 
So here I don't need to add the change to the error handling and can keep the PR focused.
Also, since changing error handling may be a bit touchy, I would prefer to do this folder change after the mime detection.
If it is OK, I can create a JIRA story for next sprint. Wdyt?

Reply:
Then you could put the new type file in the errors directory now, so it’s the same directory already.